### PR TITLE
error out when configurations do not match

### PR DESCRIPTION
**Pull request by nabuskey**
fixes: #416 

if a cluster was created with a configuration, then a different configuration is passed for the same cluster, error out.

```
./idpbuilder create
```

```
./idpbuilder create --use-path-routing
Oct 17 20:39:41 INFO Creating kind cluster logger=setup
Oct 17 20:39:41 INFO Runtime detected logger=setup provider=docker
Oct 17 20:39:41 INFO Cluster already exists logger=setup cluster=localdev
Oct 17 20:39:41 INFO Adding CRDs to the cluster logger=setup
Oct 17 20:39:41 INFO Setting up CoreDNS logger=setup
Oct 17 20:39:41 INFO Setting up TLS certificate logger=setup
Oct 17 20:39:41 ERROR Error while checking incompatible flags logger=setup err=provided command flags and existing configurations are incompatible. please recreate the cluster. existing: {Protocol:https Host:cnoe.localtest.me IngressHost:cnoe.localtest.me Port:8443 UsePathRouting:false SelfSignedCert:}, given: {Protocol:https Host:cnoe.localtest.me IngressHost:cnoe.localtest.me Port:8443 UsePathRouting:true SelfSignedCert:}
Error: :provided command flags and existing configurations are incompatible. please recreate the cluster. existing: {Protocol:https Host:cnoe.localtest.me IngressHost:cnoe.localtest.me Port:8443 UsePathRouting:false SelfSignedCert:}, given: {Protocol:https Host:cnoe.localtest.me IngressHost:cnoe.localtest.me Port:8443 UsePathRouting:true SelfSignedCert:}
```

### CODE REVIEW

File: api/v1alpha1/localbuild_types.go [hunk 1]
```diff
@@ -45,8 +45,19 @@ type PackageConfigsSpec struct {
 	CorePackageCustomization map[string]PackageCustomization `json:"packageCustomization,omitempty"`
 }
 
+// BuildCustomizationSpec fields cannot change once a cluster is created
+type BuildCustomizationSpec struct {
+	Protocol       string `json:"protocol,omitempty"`
+	Host           string `json:"host,omitempty"`
+	IngressHost    string `json:"ingressHost,omitempty"`
+	Port           string `json:"port,omitempty"`
+	UsePathRouting bool   `json:"usePathRouting,omitempty"`
```
Comment (blakeromano) [inline api/v1alpha1/localbuild_types.go:54]: I mean maybe a nit but do we ever expect to have other kind of routing where we want to make this an enum string?

File: api/v1alpha1/localbuild_types.go [hunk 2]
```diff
@@ -45,8 +45,19 @@ type PackageConfigsSpec struct {
 	CorePackageCustomization map[string]PackageCustomization `json:"packageCustomization,omitempty"`
 }
 
+// BuildCustomizationSpec fields cannot change once a cluster is created
+type BuildCustomizationSpec struct {
+	Protocol       string `json:"protocol,omitempty"`
+	Host           string `json:"host,omitempty"`
+	IngressHost    string `json:"ingressHost,omitempty"`
+	Port           string `json:"port,omitempty"`
```
Comment (blakeromano) [inline api/v1alpha1/localbuild_types.go:53]: Why are we choosing string rather then an int?

**Review by blakeromano — COMMENTED** [review]
overall looks good just some questions about types we are choosing

File: api/v1alpha1/localbuild_types.go (hunk 2 above)
Comment (nabuskey) [reply api/v1alpha1/localbuild_types.go:53]: I wanted to follow the standard package conventions. 

https://pkg.go.dev/net/url#URL.Port

File: api/v1alpha1/localbuild_types.go (hunk 1 above)
Comment (nabuskey) [reply api/v1alpha1/localbuild_types.go:54]: Hmm I honestly cannot think of other ways to route for now. Maybe this is something we can take a look at later along with the potential changes to the name of the flag. #417

File: api/v1alpha1/localbuild_types.go (hunk 2 above)
Comment (blakeromano) [reply api/v1alpha1/localbuild_types.go:53]: Ahh I didn't realize Go put ports as strings...

**Review by blakeromano — APPROVED** [review]

File: api/v1alpha1/localbuild_types.go (hunk 2 above)
Comment (nabuskey) [reply api/v1alpha1/localbuild_types.go:53]: Oh I know... It was a surprise for me too.

**Pull request merged**